src/main/java/educonnect/model/student/timetable/Day.java

```diff
@@ -1,49 +1,111 @@
 package educonnect.model.student.timetable;
 
+import static educonnect.commons.util.CollectionUtil.requireAllNonNull;
 import static educonnect.logic.parser.CliSyntax.PREFIXES_TIMETABLE_DAYS;
 import static java.util.Objects.requireNonNull;
 
 import java.time.DayOfWeek;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+import educonnect.model.student.timetable.exceptions.InvalidDurationException;
 import educonnect.model.student.timetable.exceptions.OverlapPeriodException;
 
 /**
  * Represents a day in a weekly timetable schedule.
  */
-public class Day {
+public class Day implements Comparable<Day> {
+    static final int DEFAULT_START_TIME_OF_DAY = 8; // 8 AM
+
+    static final int DEFAULT_END_TIME_OF_DAY = 22; // 10 PM
+    final ArrayList<Period> periods;
```

Comment:
Just to confirm, is it intended to be the default visibility?

Reply:
yep! its meant to be in package-private as other timetable classes will make use of it, but not outside of the package. :)